backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/IrToBitcode.kt

```diff
@@ -2261,11 +2262,11 @@ internal class CodeGeneratorVisitor(val context: Context, val lifetimes: Map<IrE
 
     private fun call(function: IrFunction, llvmFunction: LLVMValueRef, args: List<LLVMValueRef>,
                      resultLifetime: Lifetime): LLVMValueRef {
-        val exceptionHandler = if (function.hasAnnotation(RuntimeNames.filterExceptions)) {
-            functionGenerationContext.filteringExceptionHandler(currentCodeContext)
-        } else {
-            currentCodeContext.exceptionHandler
-        }
+
+        val exceptionHandler = function.descriptor.annotations.findAnnotation(RuntimeNames.filterExceptions)?.let {
```

Comment:
Using descriptors should be avoided. Also `properValue` implementation is strange, I would avoid using it too. Please try to use `function.findAnnotation(RuntimeNames.filterExceptions)?.getAnnotationValueOrNull<String>("mode")`.